proposals/low-level-struct-improvements.md

```diff
@@ -50,248 +45,226 @@ readonly ref struct Span<T>
+    // rules.
+    return new Span<int>(ref parameter);
 
-        // Okay: the arguments don't actually matter here. Literally any value 
-        // could be passed to this method and the return of it would still be 
-        // *safe-to-escape* outside the enclosing method. 
-        return CreateSpan(...);
-    }
+    // This should be legal. There is no safety issue here nor is there any compat issue
+    // as the created Span<T> only maintains references to the heap. It is always safe 
+    // to return
+    int[] array = new int[42];
+    return new Span<int>(ref array[0]);
 }
 ```
 
-This tension between allowing constructors such as `Span<T>(ref T field)` and ensuring compatibility with `ref struct` returning methods like `CreateSpan<T>` is a key pivot point in the design of `ref` fields.
+The rules we define for `ref` fields must ensure the `Span<T>` constructor properly restricts the *safe-to-escape* scope of constructed objects in the cases it captures `ref` state. At the same time it must ensure that we don't break the existing consumption rules for methods like `CreateSpan<T>`. 
+
+<a name="ref-fields-escape"></a>
 
-To do this we will change the escape rules for a constructor invocation, which today are the same as method invocation, on a `ref struct` that **directly** contains a `ref` field as follows:
+To accomplish this the span safety rules will be changed as follows. First for the constructor of a `ref struct` that directly contains a `ref field`: 
+- If the constructor is annotated with `[RefFieldEscapes]` then `ref` fields can be initialized with any value that has *ref-safe-to-escape* to the *calling method*
+- Else `ref` fields can only be initialized with known heap values.
 
-- If the constructor contains any `ref`, `out` or `in` parameters, and the arguments do not all refer to the heap, then the *safe-to-escape* of the return will be the current scope
-- Else if the constructor contains any `ref struct` parameters then the *safe-to-escape* of the return will be the current scope
-- Else the *safe-to-escape* will be the outside the method scope
+These rules ensure that the caller can understand when constructors will or will not capture parameters by `ref`. 
 
-Let's examine these rules in the context of samples to better understand their impact.
+Next the rules for method invocation will change as follows when the target method is annotated with `[RefFieldEscapes]`:
+- If the current method is not annotated with `[RefFieldEscapes]` then
+    - If all arguments have a *ref-safe-to-escape* scope of *heap* then the *safe-to-escape* scope of the return is to the *calling method*
+    - Else the *safe-to-escape* scope is to the *current method*
+- Else the *safe-to-escape* scope of the return is the minimum of 
+    - The existing *safe-to-escape* calculation for method invocation
+    - All of the *ref-safe-to-escape* values of `ref` and `in` arguments
```

Comment:
whooboy.  this last part went entirely over my head :)

Reply:
That language takes some getting used to but it's how the original span safety doc was written. I want this doc to provide a map for how these features will fit into our existing spec. Essentially what changes do we need to make to extend our existing rules.